core/src/main/scala/org/scalatra/util/Mimes.scala

```diff
@@ -1,101 +1,92 @@
 package org.scalatra.util
 
-import java.io.{ File, InputStream }
-import java.net.{ URI, URL }
-
-import eu.medsea.mimeutil.{ MimeType, MimeUtil2 }
-import eu.medsea.util.EncodingGuesser
-import org.slf4j.LoggerFactory
-
-import scala.collection.JavaConverters._
-import scala.util.control.Exception._
+import java.io.{ File, InputStream, ByteArrayInputStream }
+import java.net.{ URI, URLConnection }
 
 object Mimes {
-
   val DefaultMime = "application/octet-stream"
-  /**
-   * Sets supported encodings for the mime-util library if they have not been
-   * set. Since the supported encodings is stored as a static Set we
-   * synchronize access.
-   */
-  private def registerEncodingsIfNotSet(): Unit = {
-    synchronized {
-      if (EncodingGuesser.getSupportedEncodings.isEmpty) {
-        val enc = Set("UTF-8", "ISO-8859-1", "windows-1252", "MacRoman", EncodingGuesser.getDefaultEncoding)
-        EncodingGuesser.setSupportedEncodings(enc.asJava)
-      }
-    }
-  }
-  registerEncodingsIfNotSet
 }
 
 /**
- * A utility to help with mime type detection for a given file path or url
+ * A utility to help with mime type detection for a given file path or url.
  */
+
 trait Mimes {
 
   import org.scalatra.util.Mimes._
 
-  @transient private[this] val internalLogger = LoggerFactory.getLogger(getClass)
-
-  protected[this] def mimeUtil: MimeUtil2 = new MimeUtil2()
-  quiet { mimeUtil.registerMimeDetector("eu.medsea.mimeutil.detector.MagicMimeMimeDetector") }
-  quiet { mimeUtil.registerMimeDetector("eu.medsea.mimeutil.detector.ExtensionMimeDetector") }
-
+  /**
+   * Detects the mime type of a given Byte array.
+   *
+   * When inappropriate MIME Type can not be inferred,
+   * "text/pain" is returned.
+   *
+   * @param content The Byte array for which to detect the mime type
+   * @param fallback A fallback value in case no mime type can be found
+   */
   def bytesMime(content: Array[Byte], fallback: String = DefaultMime): String = {
-    detectMime(fallback) {
-      MimeUtil2.getMostSpecificMimeType(mimeUtil.getMimeTypes(content, new MimeType(fallback))).toString
+    // for backward compatibility...even when it is empty, "text/plain" may be good.
+    if (content.isEmpty == true) {
+      return fallback
     }
+
+    val is = new ByteArrayInputStream(content)
+    val mimeType = URLConnection.guessContentTypeFromStream(is)
+
+    if (mimeType != null) mimeType else "text/plain"
```

Comment:
Why isn't the `fallback` in the else here?

Reply:
@helfper When considering the most assumed use case as "a character string other than UTF - 8" from the contents of the existing test code, I judged that this method was the most suitable match. Although java.net.URLConnection.guessContentTypeFromStream has insufficient precision for distinguishing text data, in most cases it works well in this way. Besides, if it is a rare data type, we should explicitly specify Content-Type.